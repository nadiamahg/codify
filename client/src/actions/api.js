import axios from 'axios'

const api = axios.create({
    baseURL: 'http://localhost:3000/api',
})

export const getClasses = username => api.get(`classrooms/classes/${username}`)

const apis = {
    getClasses
}

export default apis