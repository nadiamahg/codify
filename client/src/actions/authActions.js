import axios from "axios";
import setAuthToken from "../utils/setAuthToken";
import jwt_decode from "jwt-decode";
import {
  GET_ERRORS,
  SET_CURRENT_USER,
  USER_LOADING
} from "./types";
// Register User
export const registerTeacher = (userData, history) => dispatch => {
  axios
    .post("/api/teachers/register", userData)
    .then(res => history.push("/loginTeacher")) // re-direct to login on successful register
    .catch(err =>
      dispatch({
        type: GET_ERRORS,
        payload: err.response.data
      })
    );
};

export const registerStudent = (userData, history) => dispatch => {
  axios
    .post("/api/students/register", userData)
    .then(res => history.push("/loginStudent")) // re-direct to login on successful register
    .catch(err =>
      dispatch({
        type: GET_ERRORS,
        payload: err.response.data
      })
    );
};

// Login - get user token
export const loginTeacher = userData => dispatch => {
  axios
    .post("/api/teachers/login", userData)
    .then(res => {
      // Save to localStorage
// Set token to localStorage
      const { token } = res.data;
      localStorage.setItem("jwtToken", token);
      // Set token to Auth header
      setAuthToken(token);
      // Decode token to get user data
      const decoded = jwt_decode(token);
      // Set current user
      dispatch(setCurrentUser(decoded));
    })
    .catch(err =>
      dispatch({
        type: GET_ERRORS,
        payload: err.response.data
      })
    );
};

export const loginStudent = userData => dispatch => {
  axios
    .post("/api/students/login", userData)
    .then(res => {
      // Save to localStorage
// Set token to localStorage
      const { token } = res.data;
      localStorage.setItem("jwtToken", token);
      // Set token to Auth header
      setAuthToken(token);
      // Decode token to get user data
      const decoded = jwt_decode(token);
      // Set current user
      dispatch(setCurrentUser(decoded));
    })
    .catch(err =>
      dispatch({
        type: GET_ERRORS,
        payload: err.response.data
      })
    );
};
// Set logged in user
export const setCurrentUser = decoded => {
  return {
    type: SET_CURRENT_USER,
    payload: decoded
  };
};
// User loading
export const setUserLoading = () => {
  return {
    type: USER_LOADING
  };
};
// Log user out
export const logoutUser = () => dispatch => {
  // Remove token from local storage
  localStorage.removeItem("jwtToken");
  // Remove auth header for future requests
  setAuthToken(false);
  // Set current user to empty object {} which will set isAuthenticated to false
  dispatch(setCurrentUser({}));
};