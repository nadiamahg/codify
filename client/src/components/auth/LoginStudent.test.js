import React from 'react';
import renderer from 'react-test-renderer';
import {LoginStudent} from './LoginStudent';
import { BrowserRouter} from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import { loginStudent } from "../../actions/authActions";
import { shallow, configure, mount } from 'enzyme';

configure({adapter: new Adapter()});

const props = {
	username: "",
    password: "",
	auth: {
		isAuthenticated: false,
	},
	loginStudent: loginStudent,
}


describe('Test suits for LoginStudent', () => {
	it('should match snapshot', () => {
		const component = renderer.create(<LoginStudent {...props}/>).toJSON();
    	expect(component).toMatchSnapshot();
	 });

	it('Should capture username correctly onChange', ()=> {
		const component = mount(<LoginStudent {...props}/>);
        const input = component.find('input').at(0);
        input.instance().value = 'myUsername';
  		input.simulate('change');
  		expect(component.state().username).toEqual('myUsername');
    });

    it('Should capture password correctly onChange', ()=> {
		const component = mount(<LoginStudent {...props}/>);
        const input = component.find('input').at(1);
        input.instance().value = 'myPassword';
  		input.simulate('change');
  		expect(component.state().password).toEqual('myPassword');
    });

    it('Should produce an error message on submission of empty form', ()=> {
    	
    	const component = mount(<LoginStudent {...props}/>);
		component.find('form').simulate('submit', { preventDefault () {} });
		console.log(component.debug());

    });



});
