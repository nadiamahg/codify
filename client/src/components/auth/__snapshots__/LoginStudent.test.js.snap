// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for LoginStudent should match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
    style={
      Object {
        "marginTop": "4rem",
      }
    }
  >
    <div
      className="col s8 offset-s2"
    >
      <div
        className="col s12"
        style={
          Object {
            "paddingLeft": "11.250px",
          }
        }
      >
        <h4>
          <b>
            Login
          </b>
           below
        </h4>
      </div>
      <form
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="input-field col s12"
        >
          <input
            className=""
            id="username"
            onChange={[Function]}
            type="text"
          />
          <label
            htmlFor="username"
          >
            Username
          </label>
          <span
            className="red-text"
          />
        </div>
        <div
          className="input-field col s12"
        >
          <input
            className=""
            id="password"
            onChange={[Function]}
            type="password"
            value=""
          />
          <label
            htmlFor="password"
          >
            Password
          </label>
          <span
            className="red-text"
          />
        </div>
        <div
          className="col s12"
          style={
            Object {
              "paddingLeft": "11.250px",
            }
          }
        >
          <button
            className="btn btn-large waves-effect waves-light hoverable blue accent-3"
            style={
              Object {
                "borderRadius": "3px",
                "letterSpacing": "1.5px",
                "marginTop": "1rem",
                "width": "150px",
              }
            }
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
